colour/blindness/machado2010.py

```diff
@@ -0,0 +1,271 @@
+    L, M, S = tsplit(cmfs.values)
+    d_L, d_M, d_S = np.asarray(d_LMS).astype(np.int_)
+
+    wavelengths = cmfs.wavelengths
+    area_L = np.trapz(L, wavelengths)
+    area_M = np.trapz(M, wavelengths)
+
+    alpha = lambda x: (20 - x) / 20
+
+    # Corrected equations as per:
+    # http://www.inf.ufrgs.br/~oliveira/pubs_files/
+    # CVD_Simulation/CVD_Simulation.html#Errata
+    L_a = alpha(d_L) * L + 0.96 * area_L / area_M * (1 - alpha(d_L)) * M
+    M_a = alpha(d_M) * M + 1 / 0.96 * area_M / area_L * (1 - alpha(d_M)) * L
+    # TODO: Check inconsistency with ground truth values, d_S domain seems
+    # to be [5, 59] instead of [0, 20].
+    S_a = cmfs.s_bar.clone().shift(d_S).values
+
+    LMS_a = tstack((L_a, M_a, S_a))
+    cmfs[wavelengths] = LMS_a
+
+    severity = '{0}, {1}, {2}'.format(d_L, d_M, d_S)
+    template = '{0} - Anomalous trichromacy ({1})'
+    cmfs.name = template.format(cmfs.name, severity)
+    cmfs.title = template.format(cmfs.title, severity)
+
+    return cmfs
+
+
+def anomalous_trichromacy_matrix_Machado2010(cmfs, primaries, d_LMS):
+    """
+    Computes Machado (2010) *CVD* matrix for given *LMS* cone fundamentals
+    colour matching functions and display primaries tri-spectral power
+    distributions with given :math:`\Delta_{LMS}` shift amount in nanometers to
+    simulate anomalous trichromacy.
+
+    Parameters
+    ----------
+    cmfs : LMS_ConeFundamentals
+        *LMS* cone fundamentals colour matching functions.
+    primaries : RGB_DisplayPrimaries
+        *RGB* display primaries tri-spectral power distributions.
+    d_LMS : array_like
+        :math:`\Delta_{LMS}` shift amount in nanometers as an array of ints.
+
+    Notes
+    -----
+    -   Input *LMS* cone fundamentals colour matching functions steps size is
+        expected to be 1 nanometer, non complying input will be interpolated
+        at 1 nanometer steps size.
+    -   Input :math:`\Delta_{LMS}` shift amount is in domain [0, 20].
+
+    Returns
+    -------
+    ndarray
+        Anomalous trichromacy matrix.
+
+    Examples
+    --------
+    >>> from colour import LMS_CMFS, DISPLAYS_RGB_PRIMARIES
+    >>> cmfs = LMS_CMFS.get('Stockman & Sharpe 2 Degree Cone Fundamentals')
+    >>> d_LMS = np.array([15, 0, 0])
+    >>> primaries = DISPLAYS_RGB_PRIMARIES.get('Apple Studio Display')
+    >>> anomalous_trichromacy_matrix_Machado2010(cmfs, primaries, d_LMS)  # noqa  # doctest: +ELLIPSIS
+    array([[-0.2777465...,  2.6515008..., -1.3737543...],
+           [ 0.2718936...,  0.2004786...,  0.5276276...],
+           [ 0.0064404...,  0.2592157...,  0.7343437...]])
+    """
+
+    if cmfs.shape.steps != 1:
+        cmfs = cmfs.clone().interpolate(SpectralShape(steps=1))
+
+    M_n = RGB_to_WSYBRG_matrix(cmfs, primaries)
+    cmfs_a = anomalous_trichromacy_cmfs_Machado2010(cmfs, d_LMS)
+    M_a = RGB_to_WSYBRG_matrix(cmfs_a, primaries)
+
+    return dot_matrix(np.linalg.inv(M_n), M_a)
+
+
+def cvd_matrix_Machado2010(deficiency, severity):
```

Comment:
I would like to find a better name for that definition, it basically does almost the same job than `colour.anomalous_trichromacy_matrix_Machado2010 ` definition except that it is using the pre-computed matrices from `colour.blindness.dataset.machado2010` module.